packit/upstream.py

```diff
@@ -162,16 +164,106 @@ def create_patches(
 
         return patch_list
 
-    def get_upstream_version(self):
-        return (
-            versioneers_runner.run(
-                versioneer=None,
-                package_name=self.package_config.metadata["package_name"],
-                category=None,
-            )
-            or self.specfile.get_full_version()
+    def get_latest_released_version(self) -> str:
+        """
+        Return version of the upstream project for the latest official release
+
+        :return: the version string (e.g. "1.0.0")
+        """
+        return versioneers_runner.run(
+            versioneer=None,
+            package_name=self.package_config.metadata["package_name"],
+            category=None,
         )
 
+    def get_specfile_version(self) -> str:
+        """ provide version from specfile """
+        return self.specfile.get_full_version()
+
+    def get_current_version(self) -> str:
+        """
+        Get version (hint `git describe`)
+
+        :return:
+        """
+        ver = run_command(self.package_config.current_version_command, output=True).strip()
+        logger.debug("version = %s", ver)
+        # FIXME: this might not work when users expect the dashes, but! RPM refuses dashes in version/release
+        ver = ver.replace("-", ".")
+        logger.debug("sanitized version = %s", ver)
+        return ver
+
+    def bump_spec(self, version: str = None, changelog_entry: str = None):
+        """
+        Run rpmdev-bumpspec on the upstream spec file: it enables
+        schanging version and adding a changelog entry
+
+        :param version: new version which should be present in the spec
+        :param changelog_entry: new changelog entry (just the comment)
+        """
+        cmd = ["rpmdev-bumpspec"]
+        if version:
+            # 1.2.3-4 means, version = 1.2.3, release = 4
+            cmd += ["--new", version]
+        if changelog_entry:
+            cmd += ["--comment", changelog_entry]
+        cmd.append(self.specfile_path)
+        run_command(cmd)
```

Comment:
What about using rebase_helper Spec object for that? It allows setting the new version and the changelog as well.

Reply:
I wouldn't be too surprised if rebase-helper used rpmdev-bumpspec on its own. The reason I chose to use the tool directly is that it's less likely to break - anything can happen in rebase-helper. If we use bumpspec directly, we should be more likely to be fine.

Maybe we can try to use rebase-helper and fall back to bumpspec.

Reply:
> Maybe we can try to use rebase-helper and fall back to bumpspec.

Looks good. I know that I do not like calling the commands with subprocess..;-)